src/serac/physics/thermal_conduction_functional.hpp

```diff
@@ -0,0 +1,450 @@
+// Copyright (c) 2019-2021, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * @file thermal_conduction.hpp
+ *
+ * @brief An object containing the solver for a thermal conduction PDE
+ */
+
+#pragma once
+
+#include "mfem.hpp"
+
+#include "serac/physics/base_physics.hpp"
+#include "serac/numerics/odes.hpp"
+#include "serac/numerics/stdfunction_operator.hpp"
+#include "serac/numerics/functional/functional.hpp"
+#include "serac/physics/state/state_manager.hpp"
+#include "serac/physics/materials/thermal_functional_material.hpp"
+
+namespace serac {
+
+/**
+ * @brief An object containing the solver for a thermal conduction PDE
+ *
+ * This is a generic linear thermal diffusion oeprator of the form
+ *
+ * \f[
+ * \mathbf{M} \frac{\partial \mathbf{u}}{\partial t} = -\kappa \mathbf{K} \mathbf{u} + \mathbf{f}
+ * \f]
+ *
+ *  where \f$\mathbf{M}\f$ is a mass matrix, \f$\mathbf{K}\f$ is a stiffness matrix, \f$\mathbf{u}\f$ is the
+ *  temperature degree of freedom vector, and \f$\mathbf{f}\f$ is a thermal load vector.
+ */
+template <int order, int dim>
+class ThermalConductionFunctional : public BasePhysics {
+public:
+  /// A timestep method and config for the M solver
+  struct TimesteppingOptions {
+    /// The timestepping method to be applied
+    TimestepMethod timestepper;
+
+    /// The essential boundary enforcement method to use
+    DirichletEnforcementMethod enforcement_method;
+  };
+
+  /**
+   * @brief A configuration variant for the various solves
+   * Either quasistatic, or time-dependent with timestep and M options
```

Comment:
What is `M` in this context?

Reply:
fixed the comment because this one stinks.

Reply:
I should add that *my* code comment stinks, not the github one above. 😆

Reply:
Too late.. super offended.